### ISSUE TITLE
Updating navigator wipes changes to it's xAxis

### ISSUE THREAD

**Issue by anajavi** [highcharts/highcharts#12406]
#### Expected behaviour
When doing chart.update({navigator: { .... }}), the navigator should keep changes done previously to its xAxis.

#### Actual behaviour
```javascript
var axis = chart.navigator.xAxis;
axis.update({labels:{ x: 0, y: 12 }});
axis.addPlotBand({ from: Date.UTC(2012, 6), to: Date.UTC(2014, 6), color: 'red' });
chart.update({navigator: { height: 50 }}); // labels are reset and plotband destroyed
```

This can be worked around by creating navigator with empty labels and plotbands:
```javascript
navigator: {
  xAxis: {
    labels: {},
    plotBands: []
  }
}
```
With empty initial labels and plotbands the changes are kept after chart.update({navigator...}).

Would it be possible to keep the axis instance instead of recreating it with new Axis()? That would make it way easier to use it in https://github.com/whawker/react-jsx-highcharts

#### Live demo with steps to reproduce
https://jsfiddle.net/qgd7p9st/1/

steps:
1. click button "add plotband and xaxis labels", the plotband is added and label position changes
2. click button "update navigator", the plotband disappears and labels are reset

From the code it seems, that navigator axis is recreated when navigator is updated and not all changes carry to new axis instance.

#### Product version
Highstock 7.2.1

#### Affected browser(s)

Chrome 78

**Comment by pawelfus** [comment]
Hi @anajavi 

Thanks for reporting the issue. I'm not sure if this is a bug. Correct way to update navigator's xAxis is to call: `chart.update({ navigator: { xAxis: { ... } } });`, demo: https://jsfiddle.net/BlackLabel/06mp1eLs/2/ - updates done via `navigator.xAxis.update(options)` don't reflect options in `chart.navigator.xAxis` config.

*Internal note:*
As written above, there's no connection between calling `navigator.xAxis.update(options)` and `userOptions.navigator.xAxis`. Right now we only update `userOptions.xAxis[index]` when there's a match. Otherwise, new options are ignored and not stored.

**Comment by anajavi** [comment]
Yeah, I agree that there are other ways to update the navigator axis.

However, it makes writing React wrapper quite complex.

Normally when plotband react component is added to axis, it can do so by calling axis.addPlotBand. When it is instead added to navigator axis, the chart.update({navigator:...) must be called.

The special handling for navigator spreads quite fast to all over the code.

But then again, this is a problem only for React and probably Angular wrapper libraries.

**Comment by anajavi** [comment]
Just keeping the axis instance on navigator update would go a long way here.

Currently a new axis instance is created every time the navigator is updated.

**Comment by pawelfus** [comment]
**Workaround:**
Demo: https://jsfiddle.net/BlackLabel/d764way9/

Plugin:
```js
(function(H) {
  H.wrap(
    H.Axis.prototype,
    'update',
    function(proceed, ...args) {
      const ret = proceed.apply(this, args);

      if (
        this.options.className &&
        this.options.className.indexOf('navigator') !== -1
      ) {
        this.chart.options.navigator[this.coll] = this.userOptions;
      }

      return ret;
    }

  );
})(Highcharts);
```

**Comment by KacperMadej** [comment]
a further activity occurs
